src/libs/state-history.js

```diff
@@ -3,3 +3,10 @@ export const set = newState => { window.history.replaceState(newState, '') }
 export const get = () => window.history.state || {}
 
 export const update = newState => { set({ ...get(), ...newState }) }
+
+export const clearCurrent = () => set({})
+
+export const clearAll = () => {
+  window.history.go(-window.history.length)
```

Comment:
This should probably be `1 - window.history.length`, but even so, do we know that this will work? If the user came from another site, then popping the history will unload the current page, at which point this code probably stops running.